pkg/agent/stream/client.go

```diff
@@ -180,6 +218,7 @@ func (c *streamer) Start() error {
 	if err != nil {
 		return err
 	}
+	hc.RegisterHealthcheck(util.AmplifyCentral, "central", c.healthcheck)
```

Comment:
Can the RegisterHealthCheck func be added as a field on *streamer so that it is easier to test without making an actual call to that package in a test?

Reply:
Moved the healthcheck registration as a callback which gets invoked after stream connection. We cannot register the healthcheck while creating the stream job since the registration will kick off a initial heathcheck making the central healthcheck failed and since the jobs relying on it would start to report not ready, the global pool will go in stopped state on agent initialization. So we need the initial registration done at the time of first stream connection.